Show how to make toolbars look better in high DPI

vadz opened the pull request.

vadz committed:
Use PNG icons instead of BMP and XPM ones in the toolbar sample

Update the sample for the 21st century (with just a bit of delay) and
use PNGs under all platforms instead of using BMPs under MSW and XPMs
elsewhere.

Copy PNG icons from Tango, even if we already have almost of them in
art/tango because we plan to also use "2x" versions of them in this
sample soon.

vadz committed:
Get rid of INIT_TOOL_BMP() macro in the toolbar sample

It isn't really useful any more now that all ports use wxBITMAP_PNG()
and just obfuscates the code.

No real changes.

vadz committed:
Remove unnecessary calls to wxToolBar::SetToolBitmapSize()

This is unnecessary, just as an existing comment said, so don't bother
doing it.

vadz committed:
Remove extra blank lines from png2c.py output

Adding them just seems unnecessary.

vadz committed:
Show how to use higher resolution toolbar bitmaps in high DPI

Update the toolbar sample to show how to use bigger images for the
toolbar in high DPI.

PBfordev commented on samples/toolbar/bitmaps/README.md:
```diff
@@ -0,0 +1,10 @@
+PNG icons in this directory are public domain were taken from Tango Desktop
```
```suggestion
PNG icons in this directory were taken from Tango Desktop
```

vadz committed:
Regenerate toolbar sample bitmaps in 32 and 64 pixel sizes

They're more appropriate than 16 and 32 pixels nowadays.

Also add SVG versions of the icons, from which these bitmaps were
created.

vadz committed:
Add wxHAS_IMAGE_RESOURCES defined under MSW and Mac

Unlike the existing wxHAS_IMAGES_IN_RESOURCES constant defined only
under MSW, the new one is also defined under Mac and could be defined
for the other platforms/ports later (e.g. wxQt could probably support it
too).

It's unfortunate that two very similar constants are needed, but it
doesn't seem wise to change the meaning of the existing constant, as
this would change how the commonly used wxICON() and wxBITMAP() macros
behave and would break all our own samples that use them for their frame
icon but don't embed this icon into the bundle resources under Mac.

Do change the toolbar sample to use the new constant however, as this
one does include the bitmaps it uses in its bundle under Mac.

vadz commented:
I'm going to merge this in order to be able to create a new PR using wxBitmapBundle on top of this code soon.
